thread_local_alloc.c

```diff
@@ -112,124 +118,51 @@ GC_INNER void GC_init_thread_local(GC_tlfs p)
     if (0 != GC_setspecific(GC_thread_key, p)) {
         ABORT("Failed to set thread specific allocation pointers");
     }
-    for (i = 1; i < TINY_FREELISTS; ++i) {
-        p -> ptrfree_freelists[i] = (void *)(word)1;
-        p -> normal_freelists[i] = (void *)(word)1;
-#       ifdef GC_GCJ_SUPPORT
-          p -> gcj_freelists[i] = (void *)(word)1;
-#       endif
-#       ifdef ENABLE_DISCLAIM
-          p -> finalized_freelists[i] = (void *)(word)1;
-#       endif
-    }
-    /* Set up the size 0 free lists.    */
-    /* We now handle most of them like regular free lists, to ensure    */
-    /* That explicit deallocation works.  However, allocation of a      */
-    /* size 0 "gcj" object is always an error.                          */
-    p -> ptrfree_freelists[0] = (void *)(word)1;
-    p -> normal_freelists[0] = (void *)(word)1;
+    for (i = 0; i < MAXOBJKINDS; ++i) {
+        for (j = 1; j < TINY_FREELISTS; ++j) {
+            p->freelists[i][j] = (void *)(word)1;
+        }
+        /* Set up the size 0 free lists.    */
+        /* We now handle most of them like regular free lists, to ensure    */
+        /* That explicit deallocation works.  However, allocation of a      */
+        /* size 0 "gcj" object is always an error.                          */
+#   ifdef GC_GCJ_SUPPORT
+        if (i == GC_gcj_kind) {
+          p->freelists[i][0] = ERROR_FL;
+        } else {
+#   endif
```

Comment:
change code to (with the proper indentation):
} else
\#endif
/* else */ {
p -> ...
}